app/auth/vault_auth.rb

```diff
@@ -1,74 +1,63 @@
 class VaultAuth < Auth
+  def self.vault_url
+    ENV['VAULT_URL']
+  end
+
   def self.enabled?
-    ENV['VAULT_URL'].present?
+    vault_url.present?
   end
 
   def login
-    authenticate
+    req = Net::HTTP::Post.new("/v1/auth/github/login")
+    req.body = { token: vault_token }.to_json
```

Comment:
So this is not a vault token but GitHub's personal token, right? I think getting a vault token should happen on client side so
1. Client calls Vault's `/v1/auth/github/login` with GH personal access token to get a vault token
2. Client pass the vault token to Barcelona
3. Barcelona calls Vault APIs with the passed vault token

Reply:
I see. So bcn cli grabs the token and passes it to barcelona. That makes sense. Will change

Reply:
yeah, so that's the background of this PR https://github.com/degica/barcelona/pull/615 with this meta API, cli can know what to do and where to request